### DISCUSSION TITLE
[New] - Pagination

### DISCUSSION THREAD

**Discussion by james-hayden-ellis** [Legal-and-General/canopy#1028] [Canopy requests]
**Type of request**
New

**Component, pattern or foundation?**
Component

**Nature of request**
Requesting we create a pagination component to enable an estimation of the amount of content available, and manipulation of and, navigation through, those multiple pages / steps / or similar...

![Pagination Component](https://user-images.githubusercontent.com/59871034/189696762-369d6768-9f00-4397-af0c-ff02204df230.png)

**Comment by PaulLG2020** [comment]
I am loving the simplicity of the first version, but feel we need the last option as well as when we use it on tables we could do with the extra controls. So maybe we can have some rules around hows it used for different amounts of information. So if a table has more than 300 entries, it shows the extra controls etc.

**Reply by james-hayden-ellis** [reply]
Yeah there'd either need to be the option to choose or some logic based on the number of results / rows / pages etc... as to which variation would appear for sure

**Reply by alexcanning** [reply]
Perhaps the top and bottom component serve slightly different purposes? The first one strikes me as more suitable for content rich pages where you need less distraction, whereas the second looks more functional for the self service space.

How would either of these approaches work on mobile?

**Reply by james-hayden-ellis** [reply]
Thusly...

![Pagination - Mobile](https://user-images.githubusercontent.com/59871034/190359550-d8f5d809-31d0-468c-9d73-29dda41c1d16.png)

**Reply by alexcanning** [reply]
Could I suggest pairing it back a bit for mobile? Is it necessary to have 1-2-3-4-5 buttons or could it be stripped back to a < and > and the page number?

**Reply by james-hayden-ellis** [reply]
To be honest it's been designed in a very reactive way - someone needs this, currently it's like that, can you tidy this up quickly?... -  there's been no defined specifications to design against...

**Reply by PaulLG2020** [reply]
This is the agile way…

**Reply by james-hayden-ellis** [reply]
The focus of this ticket was the search results, the pagination just came along for the ride
